ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/LedgerDB/V2.hs

```diff
@@ -573,13 +572,14 @@ getEnvSTM (LDBHandle varState) f =
 getStateRef ::
   (IOLike m, Traversable t) =>
   LedgerDBEnv m l blk ->
+  ResourceRegistry m ->
   (LedgerSeq m l -> t (StateRef m l)) ->
   m (t (StateRef m l))
-getStateRef ldbEnv project =
+getStateRef ldbEnv reg project =
   RAWLock.withReadAccess (ldbOpenHandlesLock ldbEnv) $ \() -> do
     tst <- project <$> readTVarIO (ldbSeq ldbEnv)
     for tst $ \st -> do
-      tables' <- duplicate $ tables st
+      (_, tables') <- allocate reg (\_ -> duplicate $ tables st) close
```

Comment:
Hmm, do we also want to return the `ResourceKey` here, such that the table can be closed via it? I think that would be more in line with our general push to close things via their `ResourceKey` if they were allocated within a registry. Especially if the registry is long-lived.

Reply:
While this is true, it turns out that the closing function of a handle will already track the resource key inside, so just by passing the handle we already will invoke the `release` on it.

However, for being more explicit I think your suggestion is ok. Let me implement it